extensions/2.0/Vendor/NV_materials_mdl/README.md

```diff
@@ -0,0 +1,511 @@
+
+MDL modules can be located in MDL search paths as defined in Section 2.2 and Appendix F of the [MDL Language Specification](https://github.com/NVIDIA/MDL-SDK/tree/master/doc/specification). If this is the case then the URI must have the `mdl://` scheme prefix. For example, `./my/module.mdl` and `my/module.mdl` are paths relative to the glTF file, whereas `mdl:///base.mdl` and `mdl:///vMaterials_2/Wood/Wood_Bark.mdl` are paths relative to the MDL search paths.
+
+Embedded MDL modules can be dependencies of other MDL modules. Therefore, they are required to be listed in the `modules` array in the order in which they must be loaded, such that all dependencies to these modules can be resolved.
+
+Example:
+```json
+"modules": [
+    {
+        "uri": "./my/module.mdl"
+    },
+    {
+        "uri": "mdl:///base.mdl"
+    },
+    {
+        "bufferView": 0,
+        "mimeType": "application/vnd.mdl",
+        "modulePath": "some/embedded/module.mdl"
+    }
+]
+```
+
+## Defining BSDF Measurements
+
+BSDF measurements (MBSDFs) as defined in Appendix B of the [MDL Language Specification](https://github.com/NVIDIA/MDL-SDK/tree/master/doc/specification) are listed in the `bsdfMeasurements` array. A `bsdfMeasurement` can have the following properties: 
+
+| Property | Type | Description | Required |
+|:---------|:-----|:------------|:---------|
+| `uri` | `string` | The URI (or IRI) of the MBSDF. | No |
+| `bufferView` | `integer` | The index of the bufferView that contains the MBSDF. | No |
+| `mimeType` | `string` | The media type of the bufferView's data. | No |
+| `name` | `string` | glTF name of the BSDF measurement. | No, Default: `""` |
+
+Each BSDF measurement defines either the `uri` property, which contains the URI (or IRI) of an external MBSDF file or a data-URI with embedded data, or the `bufferView` property, which points to a bufferView object that contains an MBSDF file. If the `bufferView` property is used then `mimeType` must also be defined and have the value `application/vnd.mdl-mbsdf`.
+
+Example:
+```json
+"bsdfMeasurements": [
+    {
+        "uri": "measurement.mbsdf"
+    },
+    {
+        "bufferView": 0,
+        "mimeType": "application/vnd.mdl-mbsdf"
+    }
+]
+```
+
+## Defining MDL Function Calls
+Function calls are listed in the `functionCalls` array. A `functionCall` can have the following properties: 
+
+| Property | Type | Description | Required |
+|:---------|:-----|:------------|:---------|
+| `module` | `integer` | The containing module's index. | No |
+| `functionName` | `string` | The unqualified function name. | :white_check_mark: Yes |
+| `type` | `Type` | The function's return type. | :white_check_mark: Yes |
```

Comment:
Consider using a more specific identifier for the type of the `type` property (2nd column). E.g., `functionCall.type` instead of `Type`.

Reply:
I'm not really sure how to choose a more specific identifier here since we are referring to the object type defined in `type.schema.json`. The same object type is used in function call arguments, so `functionCall.type` doesn't seem right. Any suggestions on how to handle this better? Maybe rename `type.schema.json` to `functionCall.type.schema.json`, but still use it in `functionCall.argument.schema.json`? Not sure if this is good practice or not.

Reply:
I opted to rename `type.schema.json` to `functionCall.type.schema.json` and reference `functionCall.type` for the `type` properties, as you suggested.